journal/src/main/java/io/camunda/zeebe/journal/file/SegmentsFlusher.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.journal.file;
+
+import io.camunda.zeebe.journal.JournalMetaStore;
+import java.util.Collection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class SegmentsFlusher {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SegmentsFlusher.class);
+  // no need to make this volatile if callers always acquire the write-lock beforehand
+  private long lastFlushedIndex;
+
+  SegmentsFlusher(final long lastFlushedIndex) {
+    resetLastFlushedIndex(lastFlushedIndex);
+  }
+
+  void resetLastFlushedIndex(final long lastFlushedIndex) {
+    this.lastFlushedIndex = lastFlushedIndex;
```

Comment:
:x: Shouldn't this also update the metastore. Otherwise if there was no explicit flush after this and if the node restarts, it can detect the segment as corrupted if the previous lastFlushedIndex was greater thatn reset index.

Reply:
```suggestion
    if(this.lastFlushedIndex > lastFlushedIndex) {
                // update metastore
    }
    this.lastFlushedIndex = lastFlushedIndex;
```

Reply:
I thought about that, I think it was one of the comments I had somewhere? Maybe I forgot to add it here.

In `RaftLog`, we always flush after any `deleteAfter` operation. We don't flush after reset because there isn't anything really to flush, and AFAIK we always reset to an index greater than the previous one (since resetting on snapshot). I wasn't 100% sure about that last point though, but it would be cheap.

Why I didn't add it here is that it would be performing two unnecessary I/Os in the `deleteAfter` case, i.e. call `storeLastFlushedIndex` here, then again since we have to flush as well.

But better be safe than not, and I can had some failsafe in the `MetaStore` to not write the same value more than once.